src/android/CordovaGPSLocation.java

```diff
@@ -248,4 +274,20 @@ public void requestPermissions(int requestCode)
     {
         PermissionHelper.requestPermissions(this, requestCode, permissions);
     }
+
+    public void addGPSWatch(CallbackContext callbackContext) {
+        gpsCallbackContext = callbackContext;
+        checkGPSStatus();
```

Comment:
This call must be outside of this method. `addGPSWatch` is doing more than add the callback.